api/src/services/connected-user-log.js

```diff
@@ -0,0 +1,37 @@
+const db = require('../db');
+
+const LOG_TYPE = 'connected-user-';
+const UPDATE_TIME_INTERVAL = 30 * 60 * 1000; // 30 minutes
+
+const saveLog = (user) => {
+  if (!user) {
+    const error = new Error('Error when saving log: Log Information missing');
+    return Promise.reject(error);
+  }
+  
+  const id = LOG_TYPE + user;
+  const log = {
+    user,
+    timestamp: new Date().getTime()
+  };
+
+  return db.medicLogs
+    .get(id)
+    .catch(error => {
+      if (error.status === 404) {
+        return { _id: id };
+      }
+      throw error;
+    })
+    .then(doc => {
+      if (log.timestamp - doc.timestamp < UPDATE_TIME_INTERVAL) {
+        return;
+      }
+      const logDoc = Object.assign(doc, log);
```

Comment:
Now that we're creating the object in this function I don't think we need the object.assign any more, and we can just inline the object creation to make it more obvious what is going on here, eg:

```suggestion
      const now = new Date().getTime();
      doc.user = user;
      doc.timestamp = now;
```

Then use `now` in the `if` condition above, and delete the `const log...` up above.

I prefer this because it's explicit about the properties it's updating, and it's initializing the object as late as possible.